apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -303,42 +336,109 @@ fn create_virtual_device() -> Device {
 }
 
 fn main() {
-    // Open the serial port
-    let port = serialport::new(DEVICE_PATH, DEVICE_BAUD_RATE)
-        .timeout(POLL_INTERVAL)
-        .open();
-
-    println!("Opened controller serial port at {DEVICE_PATH}");
+    log(Log {
+        event_id: "process-started".to_string(),
+        message: format!("Opened controller serial port at {DEVICE_PATH}"),
+        event_type: EventType::SystemAction,
+        ..Default::default()
+    });
+
+    let running = Arc::new(AtomicBool::new(true));
+    let r = running.clone();
+
+    if let Err(e) = ctrlc::set_handler(move || {
+        r.store(false, Ordering::SeqCst);
+    }) {
+        log(Log {
+            event_id: "error-setting-sigint-handler".to_string(),
+            message: e.to_string(),
+            event_type: EventType::SystemStatus,
+            ..Default::default()
+        });
+    }
 
     // Create virtual device for keypress events
+    log(Log {
+        event_id: "create-virtual-uinput-device-init".to_string(),
+        event_type: EventType::SystemAction,
+        ..Default::default()
+    });
     let mut device = create_virtual_device();
-    println!("Created virtual device");
 
     // Wait for virtual device to register
     thread::sleep(Duration::from_secs(1));
+    log(Log {
+        event_id: "create-virtual-uinput-device-complete".to_string(),
+        disposition: Disposition::Success,
+        event_type: EventType::SystemAction,
+        ..Default::default()
+    });
+
+    log(Log {
+        event_id: "controller-connection-init".to_string(),
+        event_type: EventType::SystemAction,
+        ..Default::default()
+    });
```

Comment:
These 2 logs are redundant atm for debugging, but IMO better to be verbose in case we change the logic later.